tests/__init__.py

```diff
@@ -34,6 +34,17 @@ def __deepcopy__(self, memo: Any) -> Any:
         raise NotImplementedError()
 
 
+class Dataframe:
+    def __init__(self) -> None:
+        pass
+
+    def __eq__(self, other: Any) -> Any:
+        return self
```

Comment:
I find `__eq__` returning self very odd. what's going on here?

Reply:
when a pandas DataFrame runs `df.__eq__(val)`, it returns a DataFrame of booleans with the same shape as the original DataFrame. Here I was trying to mimic the behavior of a df comparison resulting in the same class being returned, as this is partially what caused issues originally. perhaps this should be made clearer with a comment?

Reply:
Gotcha.
Yes, add a comment what the behavior we are mimicking is.

Reply:
i don't think so -- if we don't override eq, the comparison to string would return a bool and then not raise the expected error on the conditional. eg `if Dataframe() == "???"` would equate to `if False` (which wouldn't raise the expected error) without the eq override.